api/host.py

```diff
@@ -147,7 +147,10 @@ def delete_host_list(
     tags=None,
     filter=None,
 ):
-    if not any([display_name, fqdn, hostname_or_id, insights_id, provider_id, provider_type, tags, filter]):
+    # "unknown" in staleness means staleness has default values and is not being used as a filter parameter
+    if "unknown" in staleness and not any(
+        [display_name, fqdn, hostname_or_id, insights_id, provider_id, provider_type, registered_with, tags, filter]
+    ):
```

Comment:
I think you just need to account for whether `staleness` was provided at all, i.e.:
```suggestion
    if not any(
        [display_name, fqdn, hostname_or_id, insights_id, provider_id, provider_type, registered_with, staleness, tags, filter]
    ):
```

Reply:
In the first moment I also thought that, but the problem is that staleness will never be `None`, because the default value for the parameter is `["fresh", "stale", "unknown"]`.

I think we have 4 options here.

The easiest option would be to do `if staleness != ["fresh", "stale", "unknown"] and not any(...)`, but it wouldn't be correct, because if someone manually sets the staleness parameter to these values, we would reject his request too.

Another one would be to check if the parameter was actually provided, but I don't know how and if that can be done.

Another option would be to create separate staleness API parameter for the deletion endpoint, which wouldn't have any default values and then we can simply add `staleness` to `if not any(...)`. But we would then have to check `if staleness is None: staleness = default_values_for_staleness`.

The best option in my opinion is to finally add `stale_warning` to the default options in the staleness filter. That way we could check `if len(set(staleness) & {"fresh", "stale", "stale_warning"}) == 3` (I'm intentionally leaving `unknown` here because it doesn't do anything anyway and in my opinion it should be removed). That way we don't have to worry about someone manually setting staleness to `["fresh", "stale", "stale_warning"]`, because even if they do, they are not really filtering the hosts (as it would return all of them), so we can safely reject the request.
Also, not having `stale_warning` in the default options causes another problem and that is that if someone uses different parameter for filtering, it will not result in deleting all hosts satisfying that filter, because hosts that are in `stale_warning` state would not be deleted.

Reply:
I'm resolving this, since we hashed out the details on Thursday.